files/zh-cn/web/javascript/reference/global_objects/promise/index.md

```diff
@@ -99,148 +115,136 @@ const promiseC = promiseA.then(handleFulfilled2, handleRejected2);
 
-```html
-<!-- y.html -->
-<!DOCTYPE html>
-<iframe src="x.html"></iframe>
-<script>
-  const bound = frames[0].postMessage.bind(frames[0], "some data", "*");
-  Promise.resolve(undefined).then(bound);
-</script>
-```
-
-```html
-<!-- x.html -->
-<!DOCTYPE html>
-<script>
-window.addEventListener("message", (event) => {
-  document.querySelector("#text").textContent = "hello";
-  // 这一部分代码仅在追踪现有设置对象的浏览器中会被运行
-  console.log(event);
-}, false);
-</script>
-```
+所有这些方法都接受一个 Promise（确切地说是 [thenable](#thenable)）的[可迭代对象](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols#可迭代协议)，并返回一个新的 Promise。它们都支持子类化，这意味着它们可以在 `Promise` 的子类上调用，结果将是一个属于子类类型的 Promise。为此，子类的构造函数必须实现与 [`Promise()`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise) 构造函数相同的签名——接受一个以 `resolve` 和 `reject` 回调函数作为参数的单个 `executor` 函数。子类还必须有一个 `resolve` 静态方法，可以像 {{jsxref("Promise.resolve()")}} 一样调用，以将值解析为 Promise。
 
-在上面的示例中，`<iframe>` 仅在现有设置对象被追踪时才会被更新。这是因为在不追踪的情况下，我们可能会使用错误的环境发送消息。
-
-> **备注：** 目前，Firefox 完全实现了现有领域追踪，Chrome 和 Safari 仅部分实现。
+请注意，JavaScript 的本质上是[单线程的](/zh-CN/docs/Glossary/Thread)，因此在任何时刻，只有一个任务会被执行，尽管控制权可以在不同的 Promise 之间切换，从而使 Promise 的执行看起来是并发的。在 JavaScript 中，[并行执行](https://zh.wikipedia.org/wiki/并行计算)只能通过[工作线程](/zh-CN/docs/Web/API/Web_Workers_API)实现。
 
 ## 构造函数
 
 - {{jsxref("Promise/Promise", "Promise()")}}
-  - : 创建一个新的 `Promise` 对象。该构造函数主要用于包装还没有添加 promise 支持的函数。
+  - : 创建一个新的 `Promise` 对象。该构造函数主要用于封装还没有添加 promise 支持的函数。
+
+## 静态属性
+
+- {{jsxref("Promise/@@species", "Promise[@@species]")}}
+  - : 返回用于构造从 Promise 方法返回值的构造函数。
 
 ## 静态方法
 
-- {{jsxref("Promise.all", "Promise.all(iterable)")}}
-  - : 这个方法返回一个新的 promise 对象，等到所有的 promise 对象都成功或有任意一个 promise 失败。
+- {{jsxref("Promise.all()")}}
 
-    如果所有的 promise 都成功了，它会把一个包含 iterable 里所有 promise 返回值的数组作为成功回调的返回值。顺序跟 iterable 的顺序保持一致。
+  - : 等待所有的 Promise 被兑现，或者有任何一个被拒绝。
 
-    一旦有任意一个 iterable 里面的 promise 对象失败则立即以该 promise 对象失败的理由来拒绝这个新的 promise。
+    如果返回的 Promise 被兑现，那么它将以一个聚合数组的形式返回已兑现的 Promise 的值，顺序与包含多个 Promise 的可迭代对象中定义的顺序相同。
 
-- {{jsxref("Promise.allSettled", "Promise.allSettled(iterable)")}}
-  - : 等到所有 promise 都已敲定（每个 promise 都已兑现或已拒绝）。
+    如果有任何一个 Promise 被拒绝，那么返回的 Promise 就会被拒绝，并且拒绝的原因是可迭代对象中第一个被拒绝的 Promise 的原因。
 
-    返回一个 promise，该 promise 在所有 promise 都敲定后完成，并兑现一个对象数组，其中的对象对应每个 promise 的结果。
+- {{jsxref("Promise.allSettled()")}}
 
-- {{jsxref("Promise.any", "Promise.any(iterable)")}}
-  - : 接收一个 promise 对象的集合，当其中的任意一个 promise 成功，就返回那个成功的 promise 的值。
+  - : 等待所有 Promise 都已敲定（可能是成功兑现或被拒绝）。
 
-- {{jsxref("Promise.race", "Promise.race(iterable)")}}
-  - : 等到任意一个 promise 的状态变为已敲定。
+    返回一个 Promise，该 Promise 在所有给定的 Promise 都已成功兑现或被拒绝后被兑现，并且返回一个描述每个 Promise 结果的对象数组。
 
-    当 iterable 参数里的任意一个子 promise 成功或失败后，父 promise 马上也会用子 promise 的成功返回值或失败详情作为参数调用父 promise 绑定的相应处理函数，并返回该 promise 对象。
+- {{jsxref("Promise.any()")}}
+  - : 接受一个 Promise 对象的可迭代对象，一旦可迭代对象中的任意一个 Promise 对象成功兑现，就会返回一个新的 Promise 对象，该新的 Promise 对象兑现该 Promise的 兑现值。
```

Comment:
Actually, I don't like the desciption in en-US documation. That is, this method is not a "block method", it should return a promise immediately other than waiting for some promise is settled.

```js
console.log("start time: " + new Date().getTime())
promise = Promise.any([
	new Promise((resolve, reject) => {
		setTimeout(() => {
			resolve('foo')
		}, 3000)
	})
])

console.log("returned time: " + new Date().getTime())

promise.then((value) => {
	console.log("resolved time: " + new Date().getTime())
	console.log(value)
	// expected output: "foo"
})
```

![image](https://github.com/mdn/translated-content/assets/15844309/bb3fd274-e366-43ec-ba93-e7a7589d1562)

```suggestion
  - : 接受一个 Promise 对象的可迭代对象，一旦可迭代对象中的任意一个 Promise 对象成功兑现，就会返回一个新的 Promise 对象，该新的 Promise 对象兑现该 Promise 的兑现值。
```

Reply:
> Actually, I don't like the desciption in en-US documation. That is, this method is not a "block method", it should return a promise immediately other than waiting for some promise is settled.

I totally agree with you. I do realize that the English version has many problems, as I said in the initiation note of this PR. Considering that there is not much content in this chapter, and I need to compare all the contents of this chapter to have a deeper understanding of Promise, the modification of the original text and the update of the translation will be synchronized in the first round of all documents starting again.

Reply:
filed mdn/content#27515

Reply:
The issue has been resolved, could you update the content here @JasonLamv-t?

Reply:
> The issue has been resolved, could you update the content here @JasonLamv-t?

Sure.